frontend/app/actions/RegistrationUri.scala

```diff
@@ -0,0 +1,40 @@
+package actions
+
+import com.netaporter.uri.Uri
+import configuration.Config
+import play.api.mvc.RequestHeader
+
+object RegistrationUri {
+  val REFERRER_HEADER: String = "referer"
+  val CAMPAIGN_SOURCE: String = "MEM"
+
+  def parse(request: RequestHeader) = {
+    val redirectUrl: String = Config.idWebAppRegisterUrl(request.uri)
+    extractCampaignCode(request.headers(REFERRER_HEADER), request.path) match {
+      case Some(campaignCode) => redirectUrl.concat(s"&INTCMP=${campaignCode.get}")
+      case _ => redirectUrl
+    }
+  }
+
+  private def extractCampaignCode(referer: String, path: String) = {
+    val refererUrl = Uri.parse(referer)
+
+    val campaignReferer = refererUrl.path match {
```

Comment:
Is it possible to get these strings programatically using Play's routing mechanism? That would make things easier if URLs change in future - hopefully these raw paths would only be stored in the `routes` file